test/periphery/CardPaymentProcessorUpgradeable.test.ts

```diff
@@ -0,0 +1,1421 @@
+        someParentTxHash
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_HAS_INAPPROPRIATE_STATUS);
+
+      await expect(cardPaymentProcessor.connect(admin).confirmPayment(
+        authorizationIds[0],
+        cashOutAccount.address
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_HAS_INAPPROPRIATE_STATUS);
+
+      await expect(cardPaymentProcessor.connect(admin).confirmPayments(
+        authorizationIds,
+        cashOutAccount.address
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_HAS_INAPPROPRIATE_STATUS);
+    }
+
+    beforeEach(async () => {
+      payments = [
+        {
+          authorizationId: 1234,
+          account: user1,
+          amount: 2345,
+          status: PaymentStatus.Nonexistent,
+          makingPaymentCorrelationId: 3456,
+        },
+        {
+          authorizationId: 4567,
+          account: user2,
+          amount: 5678,
+          status: PaymentStatus.Nonexistent,
+          makingPaymentCorrelationId: 6789,
+        }
+      ]
+      admin = user2;
+      cashOutAccount = deployer;
+      authorizationIds = payments.map(payment => createBytesString(payment.authorizationId, BYTES16_LENGTH));
+      await setUpContractsForPayments(payments);
+      await addAccountToCardPaymentProcessorWhitelist(admin);
+    });
+
+    it("All payment processing functions except making are reverted if a payment was revoked", async () => {
+      await makePayments(payments);
+      await proveTx(cardPaymentProcessor.connect(admin).revokePayment(
+        authorizationIds[0],
+        someCorrelationId,
+        someParentTxHash
+      ));
+      payments[0].status = PaymentStatus.Revoked;
+      payments[0].revocationCounter = 1;
+
+      await checkCardPaymentProcessorState(payments);
+      await checkRevertingOfAllPaymentProcessingFunctionsExceptMaking();
+
+      await expect(cardPaymentProcessor.connect(payments[0].account).makePayment(
+        payments[0].amount,
+        authorizationIds[0],
+        someCorrelationId,
+      )).to.changeTokenBalances(
+        brlcMock,
+        [cardPaymentProcessor, payments[0].account],
+        [+payments[0].amount, -payments[0].amount]
+      ).and.to.emit(
+        cardPaymentProcessor,
+        "MakePayment"
+      ).withArgs(
+        authorizationIds[0],
+        someCorrelationId,
+        payments[0].account.address,
+        payments[0].amount,
+        payments[0].revocationCounter || 0
+      );
+      payments[0].status = PaymentStatus.Uncleared;
+
+      await checkCardPaymentProcessorState(payments);
+    });
+
+    it("All payment processing functions are reverted if a payment was reversed", async () => {
+      await makePayments(payments);
+      await proveTx(cardPaymentProcessor.connect(admin).reversePayment(
+        authorizationIds[0],
+        someCorrelationId,
+        someParentTxHash
+      ));
+      payments[0].status = PaymentStatus.Reversed;
+
+      await expect(cardPaymentProcessor.makePayment(
```

Comment:
Codacy has a fix for the issue: [Insert `⏎········`](https://app.codacy.com/gh/cloudwalk/brlc-token/pullRequest?prid=9781596)
```suggestion
      await expect(
```